server/src/main/java/org/elasticsearch/search/aggregations/metrics/InternalScriptedMetric.java

```diff
@@ -162,4 +162,8 @@ public int hashCode() {
         return Objects.hash(super.hashCode(), reduceScript, aggregations);
     }
 
+    @Override
+    public String toString() {
+        return "InternalScriptedMetric{" + reduceScript + "}";
```

Comment:
Had to do something here since the parent class's `toString` would call back to `toXContent` here which would throw if the instance hasn't yet been reduced.